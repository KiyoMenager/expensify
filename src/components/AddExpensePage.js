import React from 'react';

const AddExpensePage = () => (
  <div>
    This is a add expense component
  </div>
);

export default AddExpensePage;
