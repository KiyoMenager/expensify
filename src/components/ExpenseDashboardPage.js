import React from 'react';
import ExpenseList from './ExpenseList'

const ExpenseDashboardPage = () => (
  <div>
    This is a dashboard component
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
