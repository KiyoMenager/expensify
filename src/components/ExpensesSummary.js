import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectTotalExpenses from '../selectors/expenses-total';

export const ExpensesSummary = ({expensesCount, expensesTotal}) => {
  const expenseWord = expensesCount === 1 ? 'expense' : 'expenses'
  const formattedTotal = numeral(expensesTotal / 100).format('$0,0.00')
  return (
    <div>
      <h1>
        Viewing {expensesCount} {expenseWord} totalling {formattedTotal}
      </h1>
    </div>
  );
};
