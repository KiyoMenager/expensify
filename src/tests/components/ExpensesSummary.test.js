import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary with a single expense', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={345}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={2} expensesTotal={645778866}/>);
  expect(wrapper).toMatchSnapshot();
});
