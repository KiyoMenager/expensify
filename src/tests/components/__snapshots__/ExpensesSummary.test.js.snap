// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with a single expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $3.45
  </h1>
</div>
`;

exports[`should render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $6,457,788.66
  </h1>
</div>
`;
